### PR TITLE
The example does not work as noted in the README

### PR DESCRIPTION
**Pull request by jdar**
Hi Nick. Further explorations, as promised:

The procedure announced in the readme to test the library does not appear to work on a fresh install.

PART I: 
When npm calls 'node ./node_modules/ethdeploy/bin/ethdeploy.js ./ethdeploy.testrpc.config.js', node is not going to pick up meow, node-dir or other node from the current node_module folders: it's going to look in the ethdeploy node_modules. 

Likely this works on your machine because of your development linking or something.
For someone who has not done that, this would make it work:
(cd node_modules/ethdeploy && npm install)

I have added that to the README. Obviously in the long run it would be better to use the dependencies into the example node_module. For some reason specifying the node path in the example/package.json did not work:
'deploy': 'NODE_PATH=node_modules node ./node_modules/ethdeploy/bin/ethdeploy.js ./ethdeploy.testrpc.config.js'

PART II:
This part does NOT yet work in this PR. If you give me guidance I will attempt it.

After this, I had other unrelated problems running the example -- specifically with not having the existing 'input' field from environments.json deployed to testrpc. Perhaps getting these previous contracts deployed and generating a suitable config might be best acheived via docker?

### CODE REVIEW

**Comment by jdar** [comment]
Oh, and btw, the 'node_modules' location issue also affects the cli when installed thusly: 'npm install -g ethdeploy'

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Don't check auto-generated stuff into git
 node_modules
+package-lock.json
```
Comment (jdar) [inline .gitignore:3]: npm4 and below did not automatically add this. It is not useful for a dep module.

File: README.md [hunk 2]
```diff
@@ -33,12 +33,16 @@ npm install --save ethdeploy
...
+Checkout the ethdeploy [example](/example/index.js) provided. This will launch a testrpc server (a dummy ethereum api) in the background, and then deploy a bunch of contracts.
 
 ```
 cd example
 npm install
-npm run example
+(cd node_modules/ethdeploy/ && npm install)
+npm install -g eslint
+node_modules/.bin/testrpc & 
+[some initialization of the 'input' address]
```
Comment (jdar) [inline README.md]: Can you show exactly how you would do this? Or better, add a script that adds the input transaction, and spits out a new environments.json ?

**Comment by SilentCicero** [comment]
will review @jdar thanks!

File: README.md (hunk 2 above)
Comment (SilentCicero) [reply README.md]: Oh, for this I was just running testrpc, then deploying to it I believe. I should upgrade all the examples. These can all be simplified.

**Comment by SilentCicero** [comment]
@jdar due to the latest Solc, the naming conventions of the source mapped contracts has changed. I will create a filter in the solc loader to help resolve the naming, I fixed the example:

cd example
npm install
npm start

should world, and does in my environment with a fresh cloned version of ethdeploy, I will run in a VM to be sure.

I had to change the `contracts.SimpleStore` to `contracts['contracts/SimpleStore.sol:SimpleStore']` due to the new SOLC output.

Once again, I will run ethdeploy in a closed environment and see where some of these troubles are coming from. I think you are right that my development environment is skewing my perspective on node environment settings.

Will try to work this out today.

Thanks again for the input.

File: example/package.json [hunk 3]
```diff
@@ -15,9 +15,8 @@
   "author": "Nick Dodson",
   "license": "MIT",
   "dependencies": {
+    "ethdeploy": "file:../",
     "ethdeploy-solc-loader": "file:../src/loaders/solc",
     "ethdeploy-environment-loader": "file:../src/loaders/environment",
-    "ethdeploy": "file:../",
-    "ethereumjs-testrpc": "*"
```
Comment (SilentCicero) [inline example/package.json]: The example requires ethereumjs-testrpc. Not sure why this would be removed.

Comment (jdar) [reply example/package.json]: I was intending to move it to devDependencies. Will revert the change, since Dependencies is good enough.

**Pull request merged**